src/api/analytics/gtag.js

```diff
@@ -0,0 +1,192 @@
+import { links, options } from 'config'
+import formatFacetNames from 'helpers/formatFacetNames'
+import formatFilterName from 'helpers/formatFilterName'
+
+const { transformationHumanReadable: transformation } = options
+
+// adds a custom event to GA4
+// https://developers.google.com/analytics/devguides/collection/ga4/events?client_type=gtag
+const event = (eventName, value = {}, nonInteraction = false) => {
+  window.gtag('event', eventName, {
+    ...value,
+    non_interaction: nonInteraction
+  })
+}
+
+/* --- General --- */
+const emailSubscription = (from) =>
+  event('email_subscription', { subscription_click_from: from })
+
+/* --- Compendia --- */
+const compendiaDownload = (type, organism) =>
+  event(`compendia_${type}_download`, {
+    [`${type}_organism`]: organism
+  })
+
+/* --- Datasets --- */
+const myDatasetAction = (action) =>
+  event('my_dataset_action', {
+    my_dataset_action: `${action[0].toUpperCase()}${action.substring(
+      1
+    )} Samples`
+  })
+
+const myDatasetDownloadOptions = (dataset) =>
+  event('my_dataset_download_options', {
+    my_dataset_download_options: `Aggregate: ${
+      dataset.aggregate_by
+    }, Transformation: ${transformation[dataset.scale_by]}, QN: ${
+      dataset.quantile_normalize ? 'Skipped' : 'Not skipped'
+    }
+  `
+  })
+
+const datasetDownload = (token) => event('dataset_downalod', { token })
+
+const oneOffExperimentDownload = (accessionCode) =>
+  event('one_off_experiment_download', {
+    one_off_experiment_download: accessionCode
+  })
+
+const regeneratedDataset = (state, defaultOptions, newOptions) => {
+  // due to GA char limit, keys names are abbreviated
+  const format = (option) => {
+    return `A: ${option.aggregate_by}, T: ${
+      transformation[option.scale_by]
+    }, QN: ${option.quantile_normalize ? 'Not skipped' : 'Skipped'}`
+  }
```

Comment:
We should use the same logic as for `myDatasetDownloadOptions`. Use the shorter of the two if you are limited by characters.

Reply:
I've converted to this `format` method into a new helper called `getFormattedDownloadOptions` and used it in both `myDatasetDownloadOptions` and `regenerateDataset` (at 2bb98c8c735e28b65a0f113aa3984fce1af84e56).

Btw, do you prefer a shorter name? (e.g., `formatDownloadOptions` or `getDownloadOptions`)?